src/main.ts

```diff
@@ -70,18 +70,30 @@ export default class VaultChat extends Plugin {
 					new AskChatGPTModal(this.app, this, this.openAIHandler, this.getSearchResultsFiles.bind(this), indexingPromise).open();
 				}
 			});
+			
 			this.addCommand({
 				id: 'summarize-note',
 				name: 'Summarize note',
-				callback: async () => {
-					const activeFile = this.app.workspace.activeEditor?.file
-					if (activeFile) {
+				checkCallback: (checking: boolean) => {
+				const activeFile = this.app.workspace.activeEditor?.file
+				if(checking){
+					if(activeFile){
+						return  true
+					}else{
+						return false
+					}
+				}else{
+					if(activeFile){
 						const fileName = activeFile.name
-						const fileContents = await this.app.vault.read(activeFile)
-						new SummarizeNoteModal(this.app, this, this.openAIHandler, fileName, fileContents).open();
+						this.app.vault.read(activeFile).then(f => { new SummarizeNoteModal(this.app, this, this.openAIHandler, fileName, f).open()})
+						return  true
+					}else{
+						return false
 					}
 				}
+				}
```

Comment:
This could could be simplified for readability:

In general, this sort of code:
```
if(checking) {
    if(activeFile) {
        return  true
    } else {
        return false
    }
}
```
Can be simplified to just return the boolean (or return `activeFile !== undefined`, or whatever makes most sense to you):
```
if(checking) {
    return activeFile!! 
}
```

However, in this case where theres even more conditionals and return statements, the following might make more sense:
```
checkCallback: (checking: boolean) => {
	const activeFile = this.app.workspace.activeEditor?.file
	if(activeFile){
		if(!checking){
			const fileName = activeFile.name
			this.app.vault.read(activeFile).then(f => { new SummarizeNoteModal(this.app, this, this.openAIHandler, fileName, f).open()})
		}
		return true
	}
    return false
}
```